apps/site/assets/ts/schedule/components/line-diagram/__tests__/LineDiagramTest.tsx

```diff
@@ -137,4 +141,62 @@ describe("LineDiagram", () => {
     );
     expect(updateInLocationSpy).toHaveBeenCalled();
   });
+
+  it("should display the No Results card when a user doesn't query a stop", async () => {
+    render(
+      <LineDiagram
+        stopTree={stopTree}
+        route={route}
+        directionId={0}
+        alerts={[]}
+      />
+    );
+    const search = screen.getByLabelText(/Search for a */);
+    fireEvent.change(search, { target: { value: "Test Query Text" } });
+    waitFor(() => {
+      expect(screen.getByText("Test Query Text")).toBeDefined();
+      expect(screen.getByText("No stops")).toBeDefined();
+    });
+  });
+
+  it("should display the Stop Card for each stop a user queries", () => {
+    render(
+      <LineDiagram
+        stopTree={stopTree}
+        route={route}
+        directionId={0}
+        alerts={[]}
+      />
+    );
+    const search = screen.getByLabelText(/Search for a */);
+    fireEvent.change(search, { target: { value: "a" } });
+
+    // a is the stop name
+    expect(screen.getByText("a")).toBeDefined();
+  });
+
+  it("should fire the open modal event when a user clicks on the results stop card", async () => {
+    const storeHandlerSpy = jest.spyOn(store, "storeHandler");
+    render(
+      <LineDiagram
+        stopTree={stopTree}
+        route={route}
+        directionId={0}
+        alerts={[]}
+      />
+    );
+    const search = screen.getByLabelText(/Search for a */);
+    fireEvent.change(search, { target: { value: "a" } });
+
+    const scheduleButton = screen.getByText("View schedule");
```

Comment:
This test is failing because it's finding more than one button - I believe you can fix this by using `screen.queryByText` (which doesn't throw when finding more than 1 match)